corva/application.py

```diff
@@ -0,0 +1,52 @@
+from typing import Callable, List, Optional
+
+from corva.middleware.unpack_context import unpack_context
+from corva.models.base import BaseContext
+from corva.types import MIDDLEWARE_CALL_TYPE, MIDDLEWARE_TYPE
+
+
+def wrap_call_in_middleware(
+     call: Callable,
+     middleware: Optional[List[MIDDLEWARE_TYPE]] = None
+) -> MIDDLEWARE_CALL_TYPE:
+    def wrapper_factory(mw, call):
+        def wrapper(ctx):
+            return mw(ctx, call)
+
+        return wrapper
+
+    middleware = middleware or []
+
+    for mw in reversed(middleware):
+        call = wrapper_factory(mw, call)
+
+    return call
+
+
+class Corva:
+    def __init__(
+         self,
+         *,
+         middleware: Optional[List[MIDDLEWARE_TYPE[BaseContext]]] = None
```

Comment:
```suggestion
         middlewares: Optional[List[MIDDLEWARE_TYPE[BaseContext]]] = None
```

Reply:
it is used in this form everywhere I saw. maybe because of [this](https://english.stackexchange.com/a/257139)